### DISCUSSION TITLE
Built-in properties: Span, Trace - does it have any special meaning for Seq?

### DISCUSSION THREAD

**Discussion by Sebazzz** [datalust/seq-tickets#2009] [Help]
As mentioned on the [built-in properties and functions](https://docs.datalust.co/docs/built-in-properties-and-functions) documentation, Seq has some properties that are common in tracing specifically `@SpanId` and `@TraceId`. 

Does Seq currently give any specific meaning to these properties, or have any special functionality for this? Currently we use a non-standard property for this in our logging, and we're able to do tracing just fine using the query language.

**Comment by nblumhardt** [comment]
Hi @Sebazzz! Thanks for dropping us a line.

Currently there isn't very much special about `@SpanId` and `@TraceId`, but this is going to change dramatically with Seq 2024.1, which supports hierarchical trace visualization using these properties:

https://github.com/datalust/seq-tickets/issues/1986

Which client libraries are you using or planning to use to get data into Seq? I'd love to help figure out a plan/"future proof" it with you if there are any client functionality gaps to fill.

Best regards,
Nick

**Comment by Sebazzz** [comment]
> Which client libraries are you using or planning to use to get data into Seq? I'd love to help figure out a plan/"future proof" it with you if there are any client functionality gaps to fill.

We're logging with Serilog, which doesn't really have the concept of a span/trace. However, we're using Blazor (server), so we'Il scope the trace to the duration of the Blazor circuit. We're talking to the back-end through MediatR, so each time we will talk to the back-end that will be a span.

**Reply by nblumhardt** [reply]
Ah, great! Serilog is [getting first-class trace and span support](https://github.com/serilog/serilog/pull/1955) in version 3.1, which should be done within a few weeks. Your best option currently might be to wait for that version (or grab the `-dev-*` previews) and just use the out-of-the-box support.

This will work alongside any trace/span id properties that you're adding i.e. via an enricher, so the transition period should be fairly low-impact.

Hope this helps!

Nick

**Reply by Sebazzz** [reply]
Thanks for the reference to that issue!

It does speak about 'SpanId' and 'TraceId' as opposed to '@SpanId' and '@TraceId'. Is this something different or is it the same (as far as Seq is concerned)?

**Reply by nblumhardt** [reply]
Thanks for your reply. These are different - Seq treats `SpanId` and `TraceId` as regular properties, while the `@SpanId` and `@TraceId` versions are built-ins like the ones for levels, messages, and exceptions.

**Reply by Sebazzz** [reply]
I understand, but will the Seq sink for Serilog treat or rename `SpanId` or `TraceId` to `@SpanId`/`@TraceId` before passing them to Seq?

**Reply by Sebazzz** [reply]
I should rephrase my question: `@SpanId/@TraceId` will be used as an reference, but will Seq look for `TraceId` or for `trace-id` (as in the W3C spec) in the incoming logs?
